Sources/FuturedArchitecture/Architecture/ComponentModel.swift

```diff
@@ -27,5 +27,5 @@ public protocol ComponentModel: ObservableObject {
     /// The return type if this closure is `Void` intentionally. If bidirectional communication is
     /// desired, either pass closure to the *coordinator* using the event, or use other
     /// recommended pattern of data flow.
-    var onEvent: (Event) -> Void { get }
+    var onEvent: @MainActor (Event) -> Void { get }
```

Comment:
onEvent is intended to write into @MainActor bound state of the Coordinator in almost all of the cases.

With this function being defined in protocol, it is impossible to only declare @MainActor isolation at conforming type - the protocol can still be invoked and escape the isolation.